cpp/oneapi/dal/algo/kmeans/test/batch.cpp

```diff
@@ -20,22 +20,19 @@
 #include "oneapi/dal/algo/kmeans/train.hpp"
 #include "oneapi/dal/algo/kmeans/infer.hpp"
 
-#include "oneapi/dal/test/engine/common.hpp"
-#include "oneapi/dal/test/engine/dataframe.hpp"
-#include "oneapi/dal/test/engine/dataframe_math.hpp"
+#include "oneapi/dal/table/homogen.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
 #include "oneapi/dal/test/engine/fixtures.hpp"
 #include "oneapi/dal/test/engine/math.hpp"
 #include "oneapi/dal/test/engine/metrics/clustering.hpp"
-#include "oneapi/dal/table/row_accessor.hpp"
-#include "oneapi/dal/table/homogen.hpp"
 
 namespace oneapi::dal::kmeans::test {
 
 namespace te = dal::test::engine;
 namespace la = te::linalg;
 
 template <typename TestType>
-class kmeans_batch_test : public te::algo_fixture {
+class kmeans_batch_test : public te::float_algo_fixture<std::tuple_element_t<0, TestType>> {
```

Comment:
nip: At first sight, it is hard to understand what is the `0` element in the tuple `TestType`. I can only guess it is fptype.
Can we change `TestType` from `std::tuple` to some structure with meaningful type traits?

Reply:
The order depends on how types enumeration is organized in the test itself, it's not always the first element in the tuple. 

> Can we change TestType from std::tuple to some structure with meaningful type traits?

Tests often use `COMBINE_TYPES` that returns all possible combinations of types via `std::tuple`, so it's natural to use tuple as `TestType`. Adding special type trait or struct to fix this line sounds like overkill. Anyway, there are usings for tuple elements at the [next line](https://github.com/oneapi-src/oneDAL/pull/1488/files/46f7637bc4d1ed95a66889061efcdea32a644364#diff-d1d329637178f272246390e15be6186f23eb420e2362ad466b958f1bcb678cd2L40)